content/blog/2022/2022-09-02-junit5.md

```diff
@@ -0,0 +1,601 @@
+	@Test  
+	public void trueCheck() {  
+	    int dogAge = 2;  
+	    assertTrue(dogAge < 5);  
+	    System.out.println("I am young :)");  
+	}
+}
+```
+This will printout
+```
+I am young :)
+```
+### assertFalse
+It will fail if the given boolean parameters is true.
+```java
+public class DogTest { 
+	@Test  
+	public void falseCheck() {  
+	    int dogAge = 7;  
+	    assertFalse(dogAge < 5);  
+	    System.out.println("I am old :(");  
+	}
+}
+```
+This should show us
+```
+I am old :(
+```
+
+## Assertions with Hamcrest
+The first question that comes to mind is, why do we need an extra dependency to do the assertions when JUnit offers us a basic set of assertions?
+The answer is that the creators of JUnit5 intended it to be a powerful platform for running and managing tests and they didn't want to over-complicate it with a full capability of assertions. Instead, they designed the platform to be fully integrate-able with third party libraries such as Hamcrest.
+
+Hamcrest is the most popular library for such use-case, it offers a fluent API for that enable us to easily write and read object matchers.
+Let's get to know some of the most important methods of Hamcrest.
+
+First, let's introduce the Maven dependency
+```xml
+<dependency>  
+	 <groupId>org.hamcrest</groupId>  
+	 <artifactId>hamcrest-all</artifactId>  
+	 <version>1.3</version>  
+	 <scope>test</scope>  
+</dependency>
+```
+### assertThat
+This is the starting point for most of our assertion statements, as the first parameter it takes the object (or primitive) under test, and as the second parameter it takes a Matcher.
+
+```java
+import org.junit.jupiter.api.Test;  
+import static org.hamcrest.MatcherAssert.assertThat;  
+import static org.hamcrest.Matchers.*;  
+  
+public class CatTest {  
+    @Test  
+	public void testMeow() {  
+        String catName = "Stilla";  
+		int catAge = 3;  
+		boolean isNice = false;  
+  
+	   assertThat(catName, equalTo("Stilla"));  
+	   assertThat(catAge, lessThan(5));  
+	   assertThat(isNice, is(false));  
+  }  
+}
+```
+Notice the flexibility we have on the Matchers we pass, there are [many more Matchers](http://hamcrest.org/JavaHamcrest/javadoc/2.2/) we can use on many more data types.
+
+Let's explore those Matchers further more.
+
+### instanceOf
+It tests whether an object is an instance of certain class.
+Say we have a Cat class
+```java
+public class Cat {  
+}
+```
+```java
+public class CatTest {  
+	@Test  
+	public void testCatInstance() {  
+	    Cat cat = new Cat();  
+	  
+	    assertThat(cat, instanceOf(Cat.class));  
+	}
+}
+``` 
+
+### sameInstance
+It verifies if two object references are the same instance.
+```java
+public class CatTest {  
+	@Test  
+	public void testSameCatInstance() {  
+	    Cat cat = new Cat();
+	    assertThat(cat, sameInstance(cat));  
+	}
+}
+```
+
+### hasItems
+It verifies that the given items exist in the collection under test. The existence is checked through the `equals` object method.
```

Comment:
Please always add parentheses to function/method names, i.e. `equals()` in this case. This way the reader directly recognizes that its a function and not a variable name. I changed it for all instances in this article already.